src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -0,0 +1,170 @@
+import dagr.commons.CommonsDef._
+import dagr.commons.io.{Io, PathUtil}
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.util.SequenceUtil
+import htsjdk.samtools.{SAMRecord, SamReaderFactory}
+
+/**
+  * Calculates the error rate by position within each read.
+  *
+  * Gives the error rate for read one, read two, and across both reads, for the given position relative to the start of
+  * the read.  For paired end reads, the read position is relative to the first base sequenced for the given end of the
+  * read.  Soft-clipped bases, and unmapped, secondary, supplementary, duplicate, and vendor failing reads are ignored.
+  *
+  * @author Nils Homer
+  */
+@clp(description =
+  """
+    |Calculates the error rate by read position.
+  """,
+  group = ClpGroups.SamOrBam)
+class ErrorRateByReadPosition
+( @arg(flag="i", doc="Input BAM file.") val input: PathToBam,
+  @arg(flag="o", doc="Output metrics prefix. If not given, will use the input BAM basename as a prefix.")
+  val output: Option[PathPrefix] = None
+) extends FgBioTool with LazyLogging {
+  import scala.collection.JavaConversions.asScalaIterator
+
+  Io.assertReadable(input)
+  output.foreach(out => Io.assertListable(out.getParent))
+
+  def execute(): Unit = {
+    val progress = new ProgressLogger(logger, verb="written", unit=1e6.toInt)
+    val in       = SamReaderFactory.make.open(input.toFile)
+    val iterator = in.iterator()
+    val counter  = new ErrorRateByReadPositionCounter
+
+    iterator.filterNot { rec =>
+      rec.getReadUnmappedFlag || rec.isSecondaryOrSupplementary ||
+        rec.getDuplicateReadFlag || rec.getReadFailsVendorQualityCheckFlag
+    }
+    .foreach { rec =>
+      counter.accept(rec)
+      progress.record(rec)
+    }
+
+    val path = PathUtil.pathTo(output.getOrElse(PathUtil.removeExtension(input)) + ErrorRateByReadPositionMetric.FileExtension)
+    Metric.write[ErrorRateByReadPositionMetric](metrics=counter.metrics, path=path)
+
+    in.safelyClose()
+  }
+}
+
+object ErrorRateByReadPositionMetric {
+  val FileExtension = ".error_rate_by_read_position.txt"
+}
+
+/** Gives the error rate for read one, read two, and across both reads, for the given position relative to the start of
+  * the read. For paired end reads, the read position is relative to the first base sequenced for the given end of the
+  * read.
+  *
+  * @param position the read position (1-based).
+  * @param error_rate_read_one the error rate of the first read.
+  * @param error_rate_read_two the error rate of the second read.
+  * @param error_rate_total the error rate across all reads at this read position.
+  * @param bases_read_one the total number of bases observed at this position for read one.
+  * @param bases_read_two the total number of bases observed at this position for read two.
+  * @param bases_total the total number of bases observed at this position across all reads.
+  */
+case class ErrorRateByReadPositionMetric
```

Comment:
How'd you feel about changing this to have a `read_number` field and drop the `_read_one` and `_read_two` stuff - i.e. collapse the duplicate columns and have twice the rows?  I think this is awkward if read 1 and read 2 are different lengths, and honestly I don't think it's good to calculate a total error rate across the two reads for the same position in read.